src/plots/domain.js

```diff
@@ -127,6 +127,10 @@ exports.defaults = function(containerOut, layout, coerce, dfltDomains) {
         }
     }
 
-    coerce('domain.x', dfltX);
-    coerce('domain.y', dfltY);
+    var x = coerce('domain.x', dfltX);
+    var y = coerce('domain.y', dfltY);
+
+    // don't accept bad input data
+    if(!(x[0] < x[1])) containerOut.domain.x = dfltX;
```

Comment:
I would `dfltX.slice()` here. I'm sure it ever happens for traces calling `plots/domain.js`, but we do mutate the domain sometimes (e.g. when applying `scaleanchor` constraints), so I would keep things on the safe side.

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/4304/commits/74773fb1c28104c896873bf19022d6a871f0b24e